inginious/frontend/webapp/static/js/studio.js

```diff
@@ -487,6 +493,296 @@ function studio_init_template_custom(well, pid, problem)
 }
 
 /**
+ * Init a blockly template
+ * @param well: the DOM element containing the input fields
+ * @param pid
+ * @param problem
+ */
+function studio_init_template_blockly(well, pid, problem)
+{
+    var val = "";
+    if ("language" in problem) {
+        $('#language-' + pid, well).val(problem["language"]);
+    }
+    var toolbox_editor = registerCodeEditor($('#toolbox-' + pid)[0], 'xml', 10);
+    if ("toolbox" in problem) {
```

Comment:
This initialization code is far too looooong; consider using the ternary operator for giving default values:

```
toolbox_editor.setValue("toolbox" in problem ? problem["toolbox"] : '<xml xmlns="http://www.w3.org/1999/xhtml" style="display: none;"></xml>')
```